### PR TITLE
Vectorization - SIMD proposal

### PR DESCRIPTION
**Pull request by HurricanKai**

### CODE REVIEW

**Review by Perksey — COMMENTED** [review]
Pre-review session review

File: documentation/proposals/Proposal - Vectorization - SIMD.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    - For bool APIs, I felt it was better to return `Vector128<T>` with all elements set to be either all 1s (for `true`) or all 0s (for `false`) this is better for branchless code, which is generally desirable when working with SIMD.
+    - We should consider exposing a helper to check all elements / each element to be all 1s (true) or not (false)
+    - another proposal will enhance `Scalar[T]` with bitwise operations, and, if this proposal is accepted, will enhance this proposal too.
+
+# Proposed API
+To make this simpler for both me, and any poor soul that reads through all this, I've provided the API in the form it would appear in a PublicAPI.txt
+
+```cs
+Silk.NET.Maths.Vector128
+Silk.NET.Maths.Vector128<T>
```
[the remark this answers is not in the document]
Comment (HurricanKai) [reply documentation/proposals/Proposal - Vectorization - SIMD.md:30]: These aren't structs, they are static classes. Just like Scalar

File: documentation/proposals/Proposal - Vectorization - SIMD.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+static readonly Silk.NET.Maths.Vector128<T>.NegativeInfinity -> Vector128<T>
+static readonly Silk.NET.Maths.Vector128<T>.One -> Vector128<T>
+static readonly Silk.NET.Maths.Vector128<T>.Pi -> Vector128<T>
+static readonly Silk.NET.Maths.Vector128<T>.PiOver2 -> Vector128<T>
+static readonly Silk.NET.Maths.Vector128<T>.PositiveInfinity -> Vector128<T>
+static readonly Silk.NET.Maths.Vector128<T>.Tau -> Vector128<T>
+static readonly Silk.NET.Maths.Vector128<T>.Two -> Vector128<T>
+static readonly Silk.NET.Maths.Vector128<T>.Zero -> Vector128<T>
+
+static Silk.NET.Maths.Vector128.Abs<T>(Vector128<T> x) -> Vector128<T>
```
[the remark this answers is not in the document]
Comment (HurricanKai) [reply documentation/proposals/Proposal - Vectorization - SIMD.md:48]: Riiiight, didn't think of that. idk how I oversaw that, sorry.

File: documentation/proposals/Proposal - Vectorization - SIMD.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Perksey) [reply documentation/proposals/Proposal - Vectorization - SIMD.md:30]: Ah I misunderstood. Could probably do with a new name still.

**Review by Perksey — APPROVED** [review]

**Pull request merged**